astroid/nodes/node_ng.py

```diff
@@ -248,15 +262,34 @@ def parent_of(self, node):
                 return True
         return False
 
-    def statement(self):
+    @overload
+    def statement(
+        self, future: Literal[False] = ...
+    ) -> Union["nodes.Statement", "nodes.Module"]:
+        ...
+
+    @overload
+    def statement(self, future: Literal[True]) -> Union[NoReturn, "nodes.Statement"]:
+        ...
+
+    def statement(
+        self, future: bool = False
+    ) -> Union["nodes.Statement", "nodes.Module", NoReturn]:
         """The first parent node, including self, marked as statement node.
 
-        :returns: The first parent statement.
-        :rtype: NodeNG
+        TODO: Deprecate the future parameter and only raise StatementMissing and return
+        nodes.Statement
+
+        :raises StatementMissing: If no self has no parent attribute
         """
         if self.is_statement:
+            self = cast("nodes.Statement", self)
             return self
```

Comment:
```suggestion
            return cast("nodes.Statement", self)
```
The `pylint` warning is correct. In that case it should work by retuning the result of `cast` directly.
Ideally we wouldn't even need it. It might not be worth it as we risk circular dependencies, but would
`isinstance(self, nodes.Statement)` work? Might also be a breaking change if someone depends on manually assigning `is_statement`.

Reply:
It already creates a circular dependency, that's why I opted to use `cast`.

I thought about the `pylint` warning and couldn't figure out why it would be correct. `typing.cast` returns the second argument without any changes, so that should be fine, right?
I opened https://github.com/PyCQA/pylint/pull/5254 (prematurely). If the warning is indeed correct I will change that PR.